### PR TITLE
Optional pandas

### PR DESCRIPTION
**Pull request by danield137**
Still work in progress:

Done:
* made **pandas** optional on `azure-kusto-data`
* refactored a little to make code a little more readable
* extracted the `to_dataframes` method to a helpers class
 
todo:
* make pandas optional on azure-kusto-ingest as well

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/_models.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        self.convertion_funcs = {
+            "datetime": _converters.to_datetime,
+            "timespan": _converters.to_timedelta,
+            "DateTime": _converters.to_datetime,
+            "TimeSpan": _converters.to_timedelta,
+        }
+
+    @property
+    def columns_count(self):
+        return len(self.columns)
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/_models.py]: Just make sure we do not go thru the whole structure every time here

File: azure-kusto-data/azure/kusto/data/_models.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    def __iter__(self):
+        for i in range(self.columns_count):
+            yield self[i]
+
+    def __getitem__(self, key):
+        if type(key) == int:
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/_models.py]: We supported more types so far, especially long

File: azure-kusto-data/azure/kusto/data/_response.py [hunk 3]
```diff
@@ -248,18 +85,18 @@ def __iter__(self):
         return iter(self.tables)
 
     def __getitem__(self, key):
-        if isinstance(key, numbers.Number):
+        if type(key) == int:
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/_response.py]: As last comment: support long

File: azure-kusto-data/azure/kusto/data/helpers.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# TODO: not sure why this is here...
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/helpers.py]: Please remove the TODO

File: azure-kusto-data/tests/test_converter.py [hunk 5]
```diff
@@ -2,61 +2,56 @@
 
 import unittest
 from datetime import timedelta
-from azure.kusto.data._response import _KustoResultRow
+from azure.kusto.data._converters import to_datetime, to_timedelta
 
 
 class ConverterTests(unittest.TestCase):
-    """These are unit tests that should test custom converters used in."""
+    """These are unit tests that should test custom _converters used in."""
```
Comment (toshetah) [inline azure-kusto-data/tests/test_converter.py]: Remove the underscore

File: azure-kusto-data/tests/test_kusto_client.py [hunk 6]
```diff
@@ -3,16 +3,24 @@
...
 from dateutil.tz.tz import tzutc
-from pandas import DataFrame, Series
-from pandas.util.testing import assert_frame_equal
 
 from azure.kusto.data.request import KustoClient
 from azure.kusto.data.exceptions import KustoServiceError
 from azure.kusto.data._response import WellKnownDataSet
+from azure.kusto.data.helpers import dataframe_from_result_table
+
+pandas_installed = False
```
Comment (toshetah) [inline azure-kusto-data/tests/test_kusto_client.py]: Make sure that pandas is installed.

File: azure-kusto-data/azure/kusto/data/_models.py (hunk 1 above)
Comment (danield137) [reply azure-kusto-data/azure/kusto/data/_models.py]: talked about it, o(1)

**Pull request merged**